### DISCUSSION TITLE
Pertanyaan untuk Sesi Mentoring 26/1/23: Cara menata data untuk ditampilkan dalam R

### DISCUSSION THREAD

**Discussion by KZF022329** [teamalgoritma/community#842] [Q&A] [answered]
Pokok pertanyaan  Untuk Sesi Mentoring IP, Jumat, 26/01/23, ada 2:

1. Jika kita memperoleh data dr instansi yang sudah ditata dalam tampilan mereka (bukan raw data) bagaimana tipsnya agar bisa memudahkan untuk ditampilkan dalam visualisasi R?
2. Bagaimana cara mengubungkan database dengan visualisasi R agar perubahan/update dalam data base juga diikuti visualisasinya dlm R

**Comment by saskia-dwi-ulfah** [comment]
Dear Pak @KZF022329 , 

1. Agar bisa menampilkan data yang sudah rapi ke dalam visualisasi R, Anda bisa memanfaatkan `ggplot` yang sudah kita pelajari di kelas dan `ggplotly` untuk visualisasi yang lebih interaktif.
2. Untuk menghubungkan database ke R, Anda bisa melihat referensi pada [video berikut ini](https://www.youtube.com/watch?v=h_gQ470uArc&ab_channel=AlgoritmaDataScienceSchool).

Semoga jawaban ini dapat membantu

Best Regards,
Team Algoritma

**Reply by Yusraf002** [reply, marked as answer]
Dear Pak @KZF022329,

Menambahkan jawaban kak Saskia untuk pertanyaan nomor 1, Bapak tetap perlu melakukan data preparation apabila hendak melakukan pengolahan di R. hal ini dikarenakan tipe data yang sudah tersusun diluar R biasanya tetap perlu penyesuaian tipe data dan cleansing data saat akan diolah di R. 

dan untuk pertanyaan nomor 2. Agar bisa menghubungkan database dengan R kita perlu menggunakan beberapa package tertentu seperti :

1. RMySQL untuk membuat connection SQL database dengan R
2. pool atau DBI untuk melakukan fetching data dari database
untuk contoh dan penggunaannya bisa mengacu dari video yang sudah dilampirkan oleh kak Saskia, atau bisa membaca lebih detail melalui link berikut ini: https://shiny.posit.co/r/articles/build/overview/ 

sedangkan untuk membuat visual kita bisa menjadi uptodate setiap ada perubahan pada database, disini kita bisa mencoba Automate Scheduled Data Updates seperti pada contoh artikel berikut ini https://rviews.rstudio.com/2017/11/15/shiny-and-scheduled-data-r/

Semoga jawaban ini dapat membantu

Best Regards,
Team Algoritma
